freebsd/FreeBSDProcessList.c

```diff
@@ -250,6 +251,54 @@ static inline void FreeBSDProcessList_scanCPUTime(ProcessList* pl) {
       cpuData->systemAllPercent = cp_time_p[CP_SYS] + cp_time_p[CP_INTR];
       // this one is not really used
       //cpuData->idlePercent      = cp_time_p[CP_IDLE];
+
+      const int coreId = (cpus == 1) ? 0 : (i - 1);
+      if (coreId < 0)
+         continue;
+
+      // TODO: test with hyperthreading and multi-cpu systems
+      if (pl->settings->showCPUTemperature) {
+         int temperature;
+         size_t len = sizeof(temperature);
+         char mibBuffer[32];
```

Comment:
Probably worth mentioning that you need a buffer long enough for the `dev.cpu..temperature` string plus the length of the CPU number (so 21, plus 1 for NUL .. I suppose nobody has CPU identifiers running to 9 digits.

Reply:
Yes and no. You actually need to provide enough space, that even for the maximum permissible CPU number (int32_t) you can still fit this into the buffer. Thus this is intentionally a few bytes larger than it ordinarily needs to be. There are other places in `htop` over-sizing certain buffers for similar reasons. You'll get the reasoning too, if you use `-Wformat=2` with GCC, which also checks those sizes, IIRC.

Reply:
I think what I meant was "a comment here saying '32 is big enough for the fixed string plus %d expands to max. 9 digits so it is sufficient' would be nice" because I did go back to count whether the buffer was large enough.